src/backends/tensorrt/loader.cc

```diff
@@ -36,7 +36,7 @@ namespace nvidia { namespace inferenceserver {
 Status
 LoadPlan(
     const std::vector<char>& model_data, nvinfer1::IRuntime** runtime,
-    nvinfer1::ICudaEngine** engine)
+    nvinfer1::ICudaEngine** engine, int32_t dla_core_id)
 {
   // Create runtime only if it is not provided
   if (*runtime == nullptr) {
```

Comment:
Can you remove the *runtime == nullptr conditional here? It should always be true, and it seems like you should only set dla_core_id when the runtime is created but with the conditional is seems like you will actually override/change the dlacore...

Reply:
```
          // Model with dynamic shapes can't share engine, set to engine to
          // 'nullptr' as hint, but keeping runtime as it can be used repeatedly
```
According to these comments it is indeed required. If the gpu_device, dla_core_id pair is found in device_engines_, LoadPlan is not called in CreateExecutionContexts but is called in CreateExecutionContext where a new runtime need not be created but a new engine must be created. 
@tanmayv25 can you clarify my understanding is correct?

Reply:
At the very least we should only be setting the dla core on the call where the runtime is actually created, right? Why would we change the dla core of an already existing runtime?

Reply:
Agreed. I have fixed this.